backend/src/core/services/campaign.service.ts

```diff
@@ -236,6 +236,10 @@ const getCampaignDetails = async (
       'demo_message_limit',
       [literal('cred_name IS NOT NULL'), 'has_credential'],
       [literal("s3_object -> 'filename'"), 'csv_filename'],
+      [
+        literal(`s3_object ->> 'bucket' = '${config.get('aws.vaultBucket')}'`),
```

Comment:
I moved the config from `aws` to `tesseract` so all references to this config would be `tesseract.vaultBucket`

Reply:
After a successful csv upload, and then we try switching to the vault url and input the vault url, if the vault url throws an error, the frontend would show the csv upload ui if we refresh as the bucket wasn't filled as vault yet. Which may be fine to default back to the last csv successful upload, but then the error for the vault link would have to be displayed on the vault UI. Can't see a straightforward solution to this, let's think about it!

Reply:
As discussed, let's add a `temp_bucketname` field to track the type of the upload, even in the event of an error.